core/src/runtime/mod.rs

```diff
@@ -92,22 +94,59 @@ pub struct Runtime {
     pub should_report: bool,
 }
 
+/// Integer type used to count instructions and syscalls for `ExecutionReport`.
+type Count = u64;
+
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
 pub struct ExecutionReport {
-    pub instruction_counts: HashMap<Opcode, u64>,
-    pub syscall_counts: HashMap<SyscallCode, u64>,
+    pub instruction_counts: HashMap<Opcode, Count>,
```

Comment:
instruction is a bit inaccurate, it's more accurate to call it opcode_counts

instruction is an opcode + which registers to use, so its not really instruction counts.